xarray/core/missing.py

```diff
@@ -706,22 +721,140 @@ def interp_func(var, x, new_x, method, kwargs):
+            drop_axis = range(var.ndim - len(x), var.ndim)
+            new_axis = range(var.ndim - len(x), var.ndim - len(x) + new_x[0].ndim)
+            return da.map_blocks(
+                _interpnd,
+                var,
+                x,
+                new_x,
+                func,
+                kwargs,
+                dtype=var.dtype,
+                chunks=chunks,
+                new_axis=new_axis,
+                drop_axis=drop_axis,
+            )
+
+        for _x in new_x:
+            if sum([s > 1 for s in _x.shape]) > 1:
+                raise NotImplementedError(
+                    "Advanced interpolation is not implemented with chunked dimension"
+                )
+
+        # number of non interpolated dimensions
+        nconst = var.ndim - len(x)
+        const_dims = [f"dim_{i}" for i in range(nconst)]
+
+        # list of interpolated dimensions source
+        interp_dims = [_x.name for _x in x]
+
+        # mapping to the interpolated dimensions destination
+        # (if interpolating on a point, removes de dim)
+        final_dims = dict(
+            **{d: d for d in const_dims},
+            **{dim: _x.name for dim, _x in zip(interp_dims, new_x) if _x.size > 1},
+        )
+
+        # rename new_x to correspond to source dimension
+        def rename_index(index, dim):
+            if index.size == 1:
+                # a scalar has no name
+                return index
+            return IndexVariable(
+                dims=[dim],
+                data=index,
+                attrs=index.attrs,
+                encoding=index.encoding,
+                fastpath=True,
+            )
+
+        new_x = [rename_index(_x, dim) for dim, _x in zip(interp_dims, new_x)]
+
+        new_x_float = [_floatize_one_x(_x) for _x in new_x]
+        unsorted = any(
+            (np.any(np.diff(_x.data) < 0) for _x in new_x_float if _x.size > 1)
+        )
+        if unsorted:
+            warnings.warn(
+                "Interpolating to unsorted destination will rechunk the result",
+                da.PerformanceWarning,
+            )
+
+            sorted_idx = {
+                dim: _x.data.argsort()
+                for dim, _x in zip(interp_dims, new_x_float)
+                if _x.size > 1
+            }
+
+            new_x = [
+                _x[sorted_idx[dim]] if dim in sorted_idx else _x
+                for dim, _x in zip(interp_dims, new_x)
+            ]
+        # add missing dimensions to new_x
+        new_x = tuple([_x.set_dims(interp_dims) for _x in new_x])
+
+        # chunks x
+        x = tuple(
+            da.from_array(_x, chunks=chunks)
+            for _x, chunks in zip(x, var.chunks[nconst:])
+        )
+
+        # duplicate the ghost cells of the array in the interpolated dimensions
+        depth = {"quadratic": 2, "cubic": 3}.get(method, 1)
+        var, x, var_with_ghost, x_with_ghost = _add_interp_ghost(
+            var, x, nconst, depth=depth
+        )
```

Comment:
Are you sure if cubic and quadratic interpolations should work with a chunked array if we add a few edge points?
As far as I know, at least cubic spline is not a local algorithm and in principle values at a very far point can affect the interpolated result.

Reply:
In my modest research I read that the support of bspline are compact. But maybe I'm mistaken, I'll have to check more thoroughly.

Reply:
Yes, B-spline is compact, but I don't think cubic spline is.

Reply:
A quick test shows that indeed quadratic and cubic spline are not local algorithms in `scipy.interpolate.interp1d`
```python
import numpy as np
from scipy.interpolate import interp1d

x = np.linspace(0,6,20)
y = np.sin(x)
y2 = np.sin(x)
y2[10] += 1

x2 = np.linspace(0,6,100)
y3 = interp1d(x,y, kind="cubic")(x2)
y4 = interp1d(x,y2, kind="cubic")(x2)
y3-y4
```

Should I refuse to interpolate, or should I emit a warning ?

Reply:
I lean not to support inexact methods.

Reply:
ok